src/app/Table.elm

```diff
@@ -1,29 +1,95 @@
-module Table exposing (view, init, Msg)
+module Table exposing (view, init, initWithPlayersAndCup)
+
+{-|
+
+   This module handles the table of players,
+   and everything relating to that.
+
+-}
 
 import Html exposing (..)
+import Html.Events exposing (..)
+import Types exposing (PlayerID)
+import TableMsg exposing (Msg(..))
+import Dice
 
 
-{-
+type alias AppMsg =
+    Types.Msg
 
-   This module handles the view rendering for the table of players
-   and everything relating to that.
 
--}
+type alias PlayerName =
+    String
+
+
+type alias DiceCount =
+    Int
+
+
+type alias TablePlayer =
+    { id : PlayerID
+    , name : PlayerName
+    , diceCount : DiceCount
+    }
 
 
-type Model
-    = Model
+type alias TablePlayers =
+    List TablePlayer
 
 
-type Msg
-    = Msg
+type alias Model =
+    { tablePlayers : TablePlayers
+    , myCup : Dice.Cup
+    }
 
 
 init : Model
 init =
-    Model
+    { tablePlayers = noPlayers
+    , myCup = Dice.emptyCup
+    }
+
+
+initWithPlayersAndCup : List ( Int, String, Int ) -> List Int -> Model
```

Comment:
I would just simplify this to one `init` function. The current init function serves no purpose and is superseded by calling `initWithPlayersAndCup [] []`